girder/models/model_base.py

```diff
@@ -192,8 +206,13 @@ def validate(self, doc):
         :param doc: The document to validate before saving to the collection.
         :type doc: dict
         """
-        raise Exception('Must override validate() in %s model.'
-                        % self.__class__.__name__)  # pragma: no cover
+        if self.schema:
+            try:
+                jsonschema.validate(doc, self.schema)
+            except jsonschema.ValidationError as ve:
+                raise ValidationException(ve.message)
+
+        return doc
```

Comment:
One issue here is this removes the requirement to override the validate method ...